src/WebServer/Rest/UniqueMatchUriAndVerbComparer.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Restup.Webserver.Rest
+{
+    internal class UniqueMatchUriAndVerbRestControllerMethodInfoComparer : IEqualityComparer<RestControllerMethodInfo>
+    {
+        private readonly PathPartsAndParametersParsedUriComparer _parsedUriComparer;
+
+        public UniqueMatchUriAndVerbRestControllerMethodInfoComparer()
+        {
+            _parsedUriComparer = new PathPartsAndParametersParsedUriComparer();
+        }
+
+        public bool Equals(RestControllerMethodInfo x, RestControllerMethodInfo y)
+        {
+            return _parsedUriComparer.Equals(x.MatchUri, y.MatchUri) && x.Verb == y.Verb;
+        }
+
+        public int GetHashCode(RestControllerMethodInfo obj)
+        {
+            unchecked
+            {
+                return ((obj.MatchUri?.GetHashCode() ?? 0) * 397) ^ (int)obj.Verb;
```

Comment:
So it's a prime with ensures overflow of the integer as I understand. So something like HashCodeMultiplicationOverflowValue... ok that sucks, HashCodePrime will do :)